File: output/csbc_semish_2026/deepseek_deepseek_v3_2/few/python/Mina.py
import sys
import heapq

input_data = sys.stdin.read().split()
if not input_data:
    exit()

N = int(input_data[0])
grid = []
idx = 1
for i in range(N):
    row = list(map(int, input_data[idx:idx + N]))
    grid.append(row)
    idx += N

INF = 10**9
dist = [[INF] * N for _ in range(N)]
dist[0][0] = grid[0][0]  # custo inicial é 0 ou 1? se grid[0][0]==0, custo 0, se 1, custo 1 (mas início nunca bloqueado)

heap = []
heapq.heappush(heap, (dist[0][0], 0, 0))

while heap:
    d, x, y = heapq.heappop(heap)
    if d > dist[x][y]:
        continue
    
    for dx, dy in [(1,0), (-1,0), (0,1), (0,-1)]:
        nx, ny = x + dx, y + dy
        if 0 <= nx < N and 0 <= ny < N:
            new_cost = d + grid[nx][ny]
            if new_cost < dist[nx][ny]:
                dist[nx][ny] = new_cost
                heapq.heappush(heap, (new_cost, nx, ny))

print(dist[N-1][N-1])

File: output/csbc_semish_2026/deepseek_deepseek_v3_2/few/cpp/Mina.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;
    vector<vector<int>> grid(n, vector<int>(n));
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cin >> grid[i][j];
        }
    }

    vector<vector<int>> dist(n, vector<int>(n, INT_MAX));
    dist[0][0] = grid[0][0]; // custo inicial: 0 se livre, 1 se bloqueado (mas início é sempre 0)

    deque<pair<int, int>> dq;
    dq.push_front({0, 0});

    vector<pair<int, int>> dirs = {{1, 0}, {-1, 0}, {0, 1}, {0, -1}};

    while (!dq.empty()) {
        auto [x, y] = dq.front();
        dq.pop_front();

        for (auto [dx, dy] : dirs) {
            int nx = x + dx, ny = y + dy;
            if (nx < 0 || nx >= n || ny < 0 || ny >= n) continue;

            int new_cost = dist[x][y] + grid[nx][ny];
            if (new_cost < dist[nx][ny]) {
                dist[nx][ny] = new_cost;
                if (grid[nx][ny] == -1) { // não ocorre, mas mantém lógica
                    dq.push_back({nx, ny});
                } else if (grid[nx][ny] == 0) {
                    dq.push_front({nx, ny});
                } else { // grid[nx][ny] == 1
                    dq.push_back({nx, ny});
                }
            }
        }
    }

    cout << dist[n-1][n-1] << endl;
    return 0;
}